igvc_perception/src/train_eval/train.py

```diff
@@ -0,0 +1,217 @@
+                    help='size of convolution kernels/filters.')
+parser.add_argument('--lr', type=float, default=1e-3,
+                    help='learning rate.')
+parser.add_argument('--lr_decay', type=float, default=1.0,
+                    help='Learning rate decay multiplier.')
+parser.add_argument('--step_interval', type=int, default=100,
+                    help='Update learning rate every <step_interval> epochs.')
+parser.add_argument('--weight_decay', type=float, default=0.0,
+                    help='Weight decay hyperparameter.')
+
+# Other configuration.
+parser.add_argument('--save_model', action='store_true', default=False,
+                    help='Save pytorch model.')
+parser.add_argument('--save_interval', type=int, default=1,
+                    help='Save pytorch model after <save_interval> epochs.')
+parser.add_argument('--load_model', type=str, default=None,
+                    help='Load model from .pt file, either for initialization or evaluation.')
+parser.add_argument('--log_interval', type=int, default=10,
+                    help='number of batches between logging train status.')
+parser.add_argument('--vis', action='store_true', default=False,
+                    help='Visualize model output every log interval.')
+parser.add_argument('--no_cuda', action='store_true', default=False,
+                    help='disables CUDA training')
+parser.add_argument('--seed', type=int, default=1, metavar='S',
+                    help='random seed (default: 1)')
+parser.add_argument('--cfgfile', type=str, default='cfg/igvc.cfg',
+                    help='Directory containing cfg for train and evaluation.')
+parser.add_argument('--test', action='store_true', default=False,
+                    help='Skip training, and evaluate a loaded model on test data.')
+parser.add_argument('--val_samples', type=int, default=10,
+                    help='Number of validation samples to use from train data.')
+
+args = parser.parse_args()
+
+args.cuda = not args.no_cuda and torch.cuda.is_available()
+torch.manual_seed(args.seed)
+if args.cuda:
+    print('Using cuda.')
+    torch.cuda.manual_seed(args.seed)
+
+kwargs = {'num_workers': 1, 'pin_memory': True} if args.cuda else {}
+
+transform = transforms.Compose([
+                 transforms.ToTensor(),
+            ])
+
+cfg_params = utils.read_data_cfg(args.cfgfile)
+
+train_txt = cfg_params['train']
+test_txt = cfg_params['test']
+backup_dir = cfg_params['backup']
+
+if args.load_model is not None:
+    print('Loading model from %s.' % args.load_model)
+    model = torch.load(args.load_model)
+elif args.test:
+    print('Missing model file for evaluating test set.')
+    exit()
+else:
+    model = models.model.UNet(args.im_size, args.kernel_size)
+
+# Datasets and dataloaders.
+if not args.test:
+    train_dataset = IGVCDataset(train_txt, im_size=args.im_size, split='train', transform=transform, val_samples=args.val_samples)
```

Comment:
This is a really nice way to handle image datasets. Good to know!

Reply:
thanks :^)